ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -530,6 +534,41 @@ def test_ipa_dns_systemrecords_check(self):
             assert check["result"] == "SUCCESS"
             assert check["kw"]["key"] in SRV_RECORDS
 
+    def test_ipa_healthcheck_ds_ruv_check(self):
+        """
+        This test checks if RUVCheck displays the correct result.
+        """
+        suffix = ipautil.realm_to_suffix(self.master.domain.realm)
+        dn_suffix = escape_dn_chars(str(suffix))
+        conn = self.replicas[0].ldap_connect()
+        entries = conn.get_entries(
+            DN("cn=replica", "cn=" + dn_suffix, "cn=mapping tree, cn=config"),
+            filter="(objectclass=nsds5replicationagreement)",
+            attrs_list=["nsds50ruv"],
+        )
+        assert len(entries) == 1
+        if len(entries) == 1:
+            ruv_data = entries[0]["nsds50ruv"]
+        ruvs = ""
+        for el in ruv_data:
+            ruvs += el
+        replica_id_pattern = re.compile(
+            "\\d\\sldap\\:\\//[a-z]+\\.[a-z]+..........."
+        )
+        replica_id = replica_id_pattern.findall(ruvs)
+        master_ruv = replica_id[0][0]
+        replica_ruv = replica_id[1][0]
+        returncode, data = run_healthcheck(
```

Comment:
Yeah, it gets tricky because you need to keep track of all the masters, then look at all the RUVs and see if one is associated with with an unknown master or it is otherwise incomplete.